src/ASM/LR/ASMLRSpline.C

```diff
@@ -570,7 +570,7 @@ void ASMLRSpline::getNoIntPoints (size_t& nPt, size_t& nIPt)
     int ng[3] = { 0, 0, 0 };
     int nG1 = 0;
     this->getOrder(ng[0],ng[1],ng[2]);
-    for (unsigned char d = 0; d < ndim; d++)
+    for (unsigned char d = 0; d < (ndim < 3 ? ndim : 3); d++)
```

Comment:
Here also, `d < ndim && d < 3` is easier to read

Reply:
changed.